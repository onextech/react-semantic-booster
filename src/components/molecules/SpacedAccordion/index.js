import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Accordion, Icon } from 'semantic-ui-react';

const StyledDiv = styled.div`
  border: solid 1px;
  border-color: #ccc;
  border-radius: 0.5em;
  box-shadow: 1px 1px 1px 0 #ccc;
  margin: 1.5em 0;
  opacity: 0.95;
  background-color: white;
  
  &:first-child {
    margin-top: 0;
  }
  
  &:last-child {
    margin-bottom: 0;
  }
`;

class SpacedAccordion extends Component {
  state = { activeIndex: 0 }

  handleClick = (e, titleProps) => {
    const { index } = titleProps;
    const { activeIndex } = this.state;
    const newIndex = activeIndex === index ? -1 : index;

    this.setState({ activeIndex: newIndex });
  }

  render() {
    const { activeIndex } = this.state;
    const { data } = this.props;
    return (
      <Accordion fluid>
        {
          data.map((dataInfo) => {
            const { header, content, key } = dataInfo;
            return (
              <StyledDiv key={key}>
                <Accordion.Title active={activeIndex === key} index={key} onClick={this.handleClick} style={{ fontSize: '1.2em', fontWeight: 'bold', padding: '1.5em 3em' }}>
                  <Icon name='dropdown' />
                  { header }
                </Accordion.Title>
                <Accordion.Content active={activeIndex === key} style={{ padding: '0 4em 2em' }}>
                  {
                    typeof content === 'string' ?
                      <p>{content}</p> :
                      content.map((contentItem, i) => <p key={i}>{contentItem}</p>)
                  }
                </Accordion.Content>
              </StyledDiv>
            );
          })
        }
      </Accordion>
    );
  }
}

SpacedAccordion.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    header: PropTypes.string,
    content: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.arrayOf(PropTypes.string),
    ]),
  })).isRequired,
};

export default SpacedAccordion;
