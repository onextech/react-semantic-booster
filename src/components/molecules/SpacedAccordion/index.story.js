import React from 'react';
import { storiesOf } from '@storybook/react';
import SpacedAccordion from '.';

// eslint-disable-next-line import/prefer-default-export
export const faqsExample = [
  {
    key: 1,
    header: 'What is a table?',
    content: ' A table is a type of domesticated animal. Known for its loyalty and faithfulness, it can\n' +
    '            be found as a welcome guest in many households across the world.',
  },
  {
    key: 2,
    header: 'How do you acquire a table?',
    content: 'Three common ways for a prospective owner to acquire a table is from pet shops, private\n' +
    '            owners, or shelters.',
  },
  {
    key: 3,
    header: 'What kinds of tables are there?',
    content: 'There are many breeds of tables. Each breed varies in size and temperament. Owners often\n' +
    '            select a breed of table that they find to be compatible with their own lifestyle and\n' +
    '            desires from a companion.',
  },
  {
    key: 4,
    header: 'Accordion tab with multiple paragraphs (using an array of string)',
    content: [
      'Each story, and also this string, is an item in the array',
      'The Gingerbread Man - A little old woman baked a gingerbread man and when she took him out of the oven, he ran away. ' +
      'The woman and her husband chased him, as well as the pig, cow and horse. No one could catch him. ' +
      'He came to a river and a sly fox told him he could jump on his tail and he would take him across. ' +
      'He did and the fox went deeper and the gingerbread man had to jump on his back and then on his nose. ' +
      'When he got to his nose, the fox ate him. ',
      'Ugly Duckling - One duckling did not look like the others and he was sad. ' +
      'He decided to run away and came across a cottage. ' +
      'He stayed there for a while but the children scared him. ' +
      'He left and spent the winter in the pond. In the spring, he saw his reflection and realized he was a swan.',
    ],
  },
];

storiesOf('SpacedAccordion', module)
  .add('Default', () => (
    <SpacedAccordion data={faqsExample} />
  ));
