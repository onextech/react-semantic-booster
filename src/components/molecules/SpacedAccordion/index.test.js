import React from 'react';
import renderer from 'react-test-renderer';
import SpacedAccordion from '.';
import { faqsExample } from './index.story';

describe('<SpacedAccordion>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SpacedAccordion data={faqsExample} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
