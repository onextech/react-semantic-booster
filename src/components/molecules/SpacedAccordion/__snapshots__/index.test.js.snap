// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpacedAccordion> renders correctly 1`] = `
<div
  className="accordion ui fluid"
>
  <div
    className="SpacedAccordion__StyledDiv-bMoOhG gvVVzp"
  >
    <div
      className="title"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "1.2em",
          "fontWeight": "bold",
          "padding": "1.5em 3em",
        }
      }
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
      />
      What is a table?
    </div>
    <div
      className="content"
      style={
        Object {
          "padding": "0 4em 2em",
        }
      }
    >
      <p>
         A table is a type of domesticated animal. Known for its loyalty and faithfulness, it can
            be found as a welcome guest in many households across the world.
      </p>
    </div>
  </div>
  <div
    className="SpacedAccordion__StyledDiv-bMoOhG gvVVzp"
  >
    <div
      className="title"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "1.2em",
          "fontWeight": "bold",
          "padding": "1.5em 3em",
        }
      }
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
      />
      How do you acquire a table?
    </div>
    <div
      className="content"
      style={
        Object {
          "padding": "0 4em 2em",
        }
      }
    >
      <p>
        Three common ways for a prospective owner to acquire a table is from pet shops, private
            owners, or shelters.
      </p>
    </div>
  </div>
  <div
    className="SpacedAccordion__StyledDiv-bMoOhG gvVVzp"
  >
    <div
      className="title"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "1.2em",
          "fontWeight": "bold",
          "padding": "1.5em 3em",
        }
      }
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
      />
      What kinds of tables are there?
    </div>
    <div
      className="content"
      style={
        Object {
          "padding": "0 4em 2em",
        }
      }
    >
      <p>
        There are many breeds of tables. Each breed varies in size and temperament. Owners often
            select a breed of table that they find to be compatible with their own lifestyle and
            desires from a companion.
      </p>
    </div>
  </div>
</div>
`;
